### ISSUE TITLE
Route secret names can cause issues

### ISSUE THREAD

**Issue by twhiston** [tnozicka/openshift-acme#64]
Currently openshift acme expects to be able to create a secret containing the keys that is the same name as the route. However I have seen that this plays against the expectations of users, who often call their application, application secret and route by the same name, and thus causes openshift acme to fail. You do get an error in the logs, but obviously this is not exposed to anyone who does not have access to the acme namespace (read all our customers), so the result can be a little confusing.
It would be better, imo, if the secret created appended `-acme` to the name so it would be less likely to result in secret name collisions.

**Comment by tnozicka** [comment]
I think with Ingresses using Secret to hold the live TLS configuration, the name association to the Ingress (Route) make more sense as we target both for the long term.

> You do get an error in the logs, but obviously this is not exposed to anyone who does not have access to the acme namespace (read all our customers), so the result can be a little confusing.

This is a bug, it should be an event so user can see it.

>It would be better, imo, if the secret created appended -acme to the name so it would be less likely to result in secret name collisions.

Suffixes are tricky as you can potentially collide with different application, less likely, worse consequence.

My plan (nice to have) is to provide an annotation to allow you to change the synced secret name, but mainly this should be fixed by reporting an event.

**Comment by tnozicka** [comment]
Looks like we already report is as an event 
https://github.com/tnozicka/openshift-acme/blob/27f229b0a770394e9972b201330877613cc3146d/pkg/controllers/route/route.go#L626
and I've got that event now that I was testing it.

Events go away after some time - and we don't retry as thoroughly now because we don't have rate limit yet, otherwise that event would have been recreated at most on the informers resync.

**Comment by mhutter** [comment]
Concrete example where this is an issue: OpenShift consolidated logging:
- The secret {{logging-kibana}} holds a certificate for {{Subject: O=Logging, OU=OpenShift, CN=system.logging.kibana}}
- the route which exposes kibana is called {{logging-kibana}}
- I can rename neither the route nor the secret because the changes will be overwritten the next time I run {{openshift-ansible}}.

In practice I have seen many cases (including myself) where people deploy apps onto openshift and name their DC, Routes, Secrets, ... the same. I mean it makes sense, secret "app1" belongs to dc "app1" which is exposed by route "app1"...

So in short, big +1 for using a pre- or suffix (ie. {{acme.<route>}})

**Comment by synaesthetik** [comment]
Chiming in to reinforce the need for this change, especially in the context presented above by @mhutter. Due to restrictions in our environment, I am not able to adjust the way that secrets are named for any of our applications, and they will _always_ be the same name as the application/route. Mega +1 for prefixed/suffixed secret names. 😁

**Comment by jekader** [comment]
I wonder why @mhutter's patch is still not merged. We're hit by this issue since we use path-based routes on the same FQDN - since route names are unique each time this triggers the controller and causes lots of certificate requests and sometimes leads to rate limiting from letsencrypt. Having the simple patch would solve our issue as certificate data would just be taken from an existing secret instead of being generated over and over again.
